polaris/hub/client.py

```diff
@@ -408,18 +431,41 @@ def list_benchmarks(self, limit: int = 100, offset: int = 0) -> list[str]:
             A list of benchmark names in the format `owner/benchmark_name`.
         """
         with ProgressIndicator(
-            start_msg="Fetching artifacts...",
-            success_msg="Fetched artifacts.",
+            start_msg="Fetching benchmarks...",
+            success_msg="Fetched benchmarks.",
             error_msg="Failed to fetch benchmarks.",
         ):
-            # TODO (cwognum): What to do with pagination, i.e. limit and offset?
-            response = self._base_request_to_hub(
-                url="/v1/benchmark", method="GET", params={"limit": limit, "offset": offset}
-            )
-            response_data = response.json()
-            benchmarks_list = [f"{HubOwner(**bm['owner'])}/{bm['name']}" for bm in response_data["data"]]
-
-            return benchmarks_list
+            # Step 1: Fetch enough v2 benchmarks to cover the offset and limit
+            v2_json_response = self._base_request_to_hub(
+                url="/v2/benchmark", method="GET", params={"limit": limit, "offset": offset}
+            ).json()
+            v2_data = v2_json_response["data"]
+            v2_benchmarks = [f"{HubOwner(**benchmark['owner'])}/{benchmark['name']}" for benchmark in v2_data]
+
+            # If v2 benchmarks satisfy the limit, return them
+            if len(v2_benchmarks) == limit:
+                return v2_benchmarks
+
+            # Step 2: Calculate the remaining limit and fetch v1 benchmarks
+            remaining_limit = max(0, limit - len(v2_benchmarks))
+
+            v1_benchmarks = []
+            if remaining_limit > 0:
+                v1_json_response = self._base_request_to_hub(
+                    url="/v1/benchmark",
+                    method="GET",
+                    params={"limit": limit, "offset": max(0, offset - v2_json_response["metadata"]["total"])},
+                ).json()
+                v1_data = v1_json_response["data"]
+                v1_benchmarks = [
+                    f"{HubOwner(**benchmark['owner'])}/{benchmark['name']}" for benchmark in v1_data
+                ]
+
+            # Combine the v2 and v1 benchmarks
+            combined_benchmarks = v2_benchmarks + v1_benchmarks
+
+            # Ensure the final combined list respects the limit
+            return combined_benchmarks[:limit]
```

Comment:
**Suggestion**: Same here! 

- You can drop the if statement. 
- You can drop the `[:limit]` slice. 
- You don't need to initialize the `v1_benchmarks` as an empty list.